src/d/d_message.cpp

```diff
@@ -1,20 +1,1166 @@
+    { 3, 0,            FAN_HEART_GET},
+    { 4, 1,       SE_S_MSG_IMPORTANT},
+    { 5, 1,            SE_S_REACTION},
+    { 6, 1,            SE_S_MSG_GOOD},
+    { 7, 1,         SE_S_MSG_PRESAGE},
+    { 8, 0, FAN_TRANSITION_IMPACT_01},
+    { 9, 1,       SE_S_READ_RIDDLE_A},
+    {10, 1,       SE_S_READ_RIDDLE_B},
+};
+
+void dFlow_c::playSound(u32 params) {
+    if (params >= 1000) {
+        fn_80364FD0(ENEMY_SOUND_MGR, params);
+        return;
+    }
+    if (params >= 100) {
+        fn_803858D0(ENEMY_BGM_RELATED_MGR);
+        return;
+    }
+
+    s32 idx = -1;
+    for (int i = 0; i < 10; i++) {
+        if (sSoundDefs[i].mParams == params) {
+            idx = i;
+            break;
+        }
+    }
+
+    if (idx < 0) {
+        return;
+    }
+
+    switch (sSoundDefs[idx].mSoundMgr) {
+        case 0: AnotherSoundMgr__playSound(FANFARE_SOUND_MGR, sSoundDefs[idx].mSoundId); break;
+        case 1: SmallSoundManager::GetInstance()->playSound(sSoundDefs[idx].mSoundId); break;
+    }
+}
+
+bool dFlow_c::handleEventInternal(const MsbFlowInfo *element) {
+    u32 params1n2 = element->params1n2;
+    switch (element->param3) {
+        case EVENT_SET_STORYFLAG:
+            StoryflagManager::sInstance->setFlag(params1n2);
+            if (params1n2 == 0x52) {
+                dLytMeter_c::GetInstance()->setMeterField_0x13775(true);
+            }
+            if (dLytMsgWindow_c::fn_800D7B40() != 50013 && dLytMsgWindow_c::fn_800D7B40() != 20061) {
+                if (params1n2 == 100 || params1n2 == 64 || params1n2 == 271 || params1n2 == 81 || params1n2 == 668 ||
+                    params1n2 == 669) {
+                    FileManager::GetInstance()->setDowsingSlotIdx(DowsingTarget::SLOT_STORY_EVENT);
+                } else if (params1n2 == 106 || params1n2 == 107) {
+                    FileManager::GetInstance()->setDowsingSlotIdx(DowsingTarget::SLOT_QUEST);
+                }
+            }
+            break;
+        case EVENT_UNSET_STORYFLAG: StoryflagManager::sInstance->unsetFlag(params1n2); break;
+        case EVENT_SET_SCENEFLAG:
+            dStageMgr_c::GetInstance()->getFlagIndex();
+            SceneflagManager::sInstance->setFlag(0x3F, (params1n2 >> 16) & 0xFFFF);
+            break;
+        case EVENT_UNSET_SCENEFLAG:
+            dStageMgr_c::GetInstance()->getFlagIndex();
+            SceneflagManager::sInstance->unsetFlag(0x3F, (params1n2 >> 16) & 0xFFFF);
+            break;
+        case EVENT_SET_ZONEFLAG:
+            dStageMgr_c::GetInstance()->getFlagIndex();
+            SceneflagManager::sInstance->setFlag(
+                dStage_c::GetInstance()->getCurrRoomId(), ((params1n2 >> 16) & 0xFFFF) + 0xC0
+            );
+            break;
+        case EVENT_UNSET_ZONEFLAG:
+            dStageMgr_c::GetInstance()->getFlagIndex();
+            SceneflagManager::sInstance->unsetFlag(
+                dStage_c::GetInstance()->getCurrRoomId(), ((params1n2 >> 16) & 0xFFFF) + 0xC0
```

Comment:
I notice the `+ 0xC0` here for the Zone flag and the `+ 0x80` for the  Temp flag ones at line 409/413
Do we know if this offset is defined in the flag managers or is this something custom to this?

Reply:
The SceneflagManager internally uses these, so there's a chance these are inlines. Will try converting these to inlines